cmd/pctl/install.go

```diff
@@ -88,24 +99,40 @@ func installCmd() *cli.Command {
 
 // install runs the install part of the `install` command.
 func install(c *cli.Context) error {
-	profilePath, catalogClient, err := parseArgs(c)
-	if err != nil {
-		_ = cli.ShowCommandHelp(c, "install")
-		return err
+	var (
+		err           error
+		catalogClient *client.Client
+		profilePath   string
+		version       string
+		catalogName   string
+		profileName   string
+	)
+
+	// only set up the catalog if a url is not provided
+	url := c.String("url")
+	if url == "" {
+		profilePath, catalogClient, err = parseArgs(c)
+		if err != nil {
+			_ = cli.ShowCommandHelp(c, "install")
+			return err
+		}
+		parts := strings.Split(profilePath, "/")
+		if len(parts) < 2 {
+			_ = cli.ShowCommandHelp(c, "install")
+			return errors.New("both catalog name and profile name must be provided")
+		}
+		if len(parts) == 3 {
+			version = parts[2]
+		}
+		catalogName, profileName = parts[0], parts[1]
 	}
 
 	branch := c.String("branch")
 	subName := c.String("subscription-name")
 	namespace := c.String("namespace")
 	configValues := c.String("config-secret")
 	dir := c.String("out")
-
-	parts := strings.Split(profilePath, "/")
-	if len(parts) < 2 {
-		_ = cli.ShowCommandHelp(c, "install")
-		return errors.New("both catalog name and profile name must be provided")
-	}
-	catalogName, profileName := parts[0], parts[1]
+	path := c.String("path")
```

Comment:
is this mandatory when url/branch are set?

Reply:
It is otherwise which profile are you installing? Url and branch aren't enough to determine that.